rust/src/lib.rs

```diff
@@ -1,3 +1 @@
-pub mod cache_client;
-pub mod control_client;
-pub mod auth;
+include!(concat!(env!("OUT_DIR"), "/mod.rs"));
```

Comment:
Are you aware of `tonic::include_proto`?

```suggestion
pub mod auth {
    tonic::include_proto!("auth");
}
pub mod cache_client {
    tonic::include_proto!("cache_client");
}
pub mod control_client {
    tonic::include_proto!("control_client");
}
```

This would let you remove lib.tmpl.rs as well as the mirrored workaround in build.rs, while more closely matching the way tonic examples are shaped.

Reply:
I was not! That makes things way cleaner. I have rewritten everything to use `include_proto!` and dropped the extra bits from the build script.